sound/soc/sof/sof-client.h

```diff
@@ -0,0 +1,79 @@
+/* SPDX-License-Identifier: (GPL-2.0-only) */
+
+#ifndef __SOUND_SOC_SOF_CLIENT_H
+#define __SOUND_SOC_SOF_CLIENT_H
+
+#include <linux/completion.h>
+#include <linux/debugfs.h>
+#include <linux/device.h>
+#include <linux/device/driver.h>
+#include <linux/kernel.h>
+#include <linux/list.h>
+#include <linux/ancillary_bus.h>
```

Comment:
Can I get an SOF developer to comment on this?

Reply:
clean up most of headers

Reply:
From Documentation/process/submit-checklist.rst:

> 1) If you use a facility then #include the file that defines/declares
>    that facility.  Don't depend on other header files pulling in ones
>    that you use.

So, no, you have to include headers directly or forward-declare structures. E.g. you do have to `#include <linux/list.h>` here. There are some exceptions like you usually `#include <linux/bitops.h>` instead of `#include <asm/bitops.h>` for `ffs()`, `ffz()`, etc. but usually you do include respective headers directly.

Reply:
headers still missing